lldb/include/lldb/DataFormatters/TypeSummary.h

```diff
@@ -258,6 +258,9 @@ class TypeSummaryImpl {
 
   virtual std::string GetDescription() = 0;
 
+  virtual ConstString GetName() = 0;
+  virtual ConstString GetImplType() = 0;
```

Comment:
Could we use something more descriptive for the name? My expectation from a function called `GetImplType` is that it returns something other than a string, for example, an enum. So maybe something along the lines of `GetSummaryFormatKindName`?

Reply:
@Michael137 I also prefer `KindName`, but there is also an existing `Kind` enum used in TypeSummary, so I didn't want to cause confusion, any other ideas?

Reply:
I ended up just going with kind name